client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx

```diff
@@ -217,13 +280,68 @@ const PlanOnlyThankYou = ( {
 					/>
 				}
 				footerDetails={ footerDetails }
+				upsellProps={ upsellProps }
 			/>
 		</>
 	);
 };
 
-export default connect( null, {
-	errorNotice,
-	removeNotice,
-	successNotice,
-} )( PlanOnlyThankYou );
+function getBusinessPlanSlugToUpgrade( currentPlanSlug: string ) {
+	const currentPlan = getPlan( currentPlanSlug );
+	const planKey = findFirstSimilarPlanKey( PLAN_BUSINESS, { term: currentPlan?.term } );
+
+	// It's just for TS, since .find() can return undefined
+	if ( ! planKey ) {
+		return;
+	}
+
+	return getPlan( planKey );
+}
+
+export default connect(
+	( state: IAppState, props: PlanOnlyProps ) => {
+		let promotePlanData;
+		if ( isPersonal( props.primaryPurchase ) || isPremium( props.primaryPurchase ) ) {
+			const promotePlan = getBusinessPlanSlugToUpgrade( props.primaryPurchase.productSlug );
```

Comment:
We may consider renaming that function - its name is misleading since it returns an object rather than a slug. Actually, we should also try to use a name for the variable that is more consistent:

```suggestion
			const businessPlan = getBusinessPlanToUpgrade( props.primaryPurchase.productSlug );
```

Reply:
`getBusinessPlanToUpgrade ` - agree
`businessPlan ` - I would argue :)  `PlanOnlyThankYou` expects any plan to promote, so I would be abstract, to make it more flexible for the future, in case if we need to promote another plan or something else.

Reply:
I'm sensing a recurring theme 😅. I don't think we should optimize our code for future changes when that future is unknown. We're sacrificing immediate readability for possible flexibility - it would be easy enough to rename this variable if we really need to add support for another plan. Again, if we indeed have concrete plans to promote more plans in another pull request, then by all means, keep it as is 🙂.

Reply:
Oh, this variable is used only inside the `connect` callback and here I totally agree that it's better to be more specific with naming, sorry :) Changed the name 👍